master/internal/configpolicy/task_config_policy.go

```diff
@@ -160,27 +161,90 @@ func GetMergedConstraints(ctx context.Context, workspaceID int, workloadType str
 	return &constraints, nil
 }
 
-// PriorityAllowed returns true if the desired priority is within the task config policy limit.
-func PriorityAllowed(wkspID int, workloadType string, priority int, smallerHigher bool) (bool, error) {
+func findAllowedPriority(scope *int, workloadType string) (limit int, found bool, err error) {
```

Comment:
```suggestion
func findAllowedPriority(scope *int, workloadType string) (limit int, constraintFound bool, err error) {
```
I think this change would make the function a lot clearer! 
Alternatively, can we include a comment saying that `found` implies a priority constraint was found?
And that we return `false` when priority is set in an invariant config because it is accepted that invariant configs adhere to constraints (so if an invariant config is set, it can't possible violate a constraint)?
i think this logic is clear to us since we've worked on the project, but could be interpreted as a bug without this context.

Reply:
I also think changing the function name to `findEnforcedPriority` would be helpful! the name `findAllowedPriority` (to me) implies that the rules of accepting or rejecting a requested priority are the same for invariant configs and constraints, which isn't the case (if priority is set in the invariant config, any priority request that varies from it is rejected, whereas for constraints, priority request below it is accepted)

Reply:
can you rename `found`  to `exists`? that sounds more familiar